src/process/namespaces.rs

```diff
@@ -12,22 +13,27 @@ use super::Process;
 impl Process {
     /// Describes namespaces to which the process with the corresponding PID belongs.
     /// Doc reference: https://man7.org/linux/man-pages/man7/namespaces.7.html
-    pub fn namespaces(&self) -> ProcResult<Vec<Namespace>> {
+    /// The namespace type is the key for the HashMap, i.e 'net', 'user', etc.
+    pub fn namespaces(&self) -> ProcResult<HashMap<OsString, Namespace>> {
         let ns = self.root.join("ns");
-        let mut namespaces = Vec::new();
+        let mut namespaces = HashMap::new();
         for entry in fs::read_dir(ns)? {
             let entry = entry?;
             let path = entry.path();
             let ns_type = entry.file_name();
             let stat = rustix::fs::statat(&rustix::fs::cwd(), &path, AtFlags::empty())
                 .map_err(|_| build_internal_error!(format!("Unable to stat {:?}", path)))?;
 
-            namespaces.push(Namespace {
-                ns_type,
-                path,
-                identifier: stat.st_ino,
-                device_id: stat.st_dev,
-            })
+            namespaces
+                .insert(
+                    ns_type,
+                    Namespace {
+                        path,
+                        identifier: stat.st_ino,
+                        device_id: stat.st_dev,
+                    },
+                )
+                .expect("NsType already in map");
```

Comment:
Sure will fix the expect with a proper error propagation, I'm sorry I'm not sure I understand which part is backwards?
We insert to the HashMap, using the ns_type as Key, and the Value is the Namespace structure.

Reply:
Take a look at the CI test failure.

The first time we insert something into the HashMap, it will return `None` to indicate that the key doesn't exist in the map already.  But this causes the `expect` to panic

Reply:
Oh I see thank you sorry! missed something silly like that will fix :)